pandas/io/tests/test_parsers.py

```diff
@@ -2623,7 +2615,145 @@ def test_eof_states(self):
                           StringIO(data), escapechar='\\')
 
 
-class TestPythonParser(ParserTests, tm.TestCase):
+class CompressionTests(object):
+    def test_zip(self):
+        try:
+            import zipfile
+        except ImportError:
+            raise nose.SkipTest('need zipfile to run')
+
+        with open(self.csv1, 'rb') as data_file:
+            data = data_file.read()
```

Comment:
you don't really need to make this with encompass the entire block

```
with open(self.csv1, 'rb') as data_file:
    data = data_file.read()
expected = .....

with tm.ensure_clean() as path:
     file_name = ....
     ....
     tmp.writestr(file_name, data)

```

as ``data`` is in scope